src/lib/parts.ts

```diff
@@ -37,6 +37,61 @@ export const isIterable = (value: unknown): value is Iterable<unknown> => {
       !!(value && (value as any)[Symbol.iterator]);
 };
 
+/**
+ * A Part that controls all or part of an attribute value.
+ */
+export class SpreadPart implements Part {
+  value: unknown = undefined;
+  subparts: Map<string, ReadonlyArray<Part>> = new Map();
+  element: Element;
+  handleAttribute: (name: string) =>  ReadonlyArray<Part>;
+  // Names of attributes that were spread in the last commit,
+  // but are not anymore.
+  private __removedNames: Array<string> = [];
+
+  constructor(element: Element, handleAttribute: (name: string) =>  ReadonlyArray<Part>) {
+    this.element = element;
+    this.handleAttribute = handleAttribute;
+  }
+
+  setValue(value: unknown): void {
+    let spreadable = value as {[key: string]: any};
+    for (var name in spreadable) {
+      this.subparts.set(name, this.handleAttribute(name))
+    }
+    for (var [name, parts] of this.subparts.entries()) {
+      if (parts) {
```

Comment:
We try to be very explicit with conditionals, partially for clarity and partially to avoid ToBoolean conversions.

In this case I want to know what you're checking for, that `parts` is not `undefined, or that it's not empty. Either way are we sure the check is necessary?

Reply:
You're right that this check isn't necessary.